ananse/peakpredictor.py

```diff
@@ -207,6 +219,117 @@ def _jaccard_motif_graph(self, indirect, factors):
+
+        Parameters
+        ----------
+        regions : list, optional
+            list of regions to filter the pfmscorefile by (Not yet supported)
+        pfmscorefile : str/file
+            pre-scanned gimmemotifs scores file (Not yet supported)
+        cage_tpms : str
+            file should contain bidirectional regions and TPMs
+        """
+        # error checking
+        if not os.path.exists(cage_tpms):
+            raise FileNotFoundError(f"Could not find {cage_tpms}")
+        elif regions is not None:
+            raise ValueError("Using custom regions with CAGE input is currently not supported.")
+        else:
+
+            # load cage tpms
+            data = pd.read_table(cage_tpms, sep="\t", comment="#")
+            if len(data.columns) != 2:
+                raise ValueError(
+                    "For the CAGE TPM file, please give only two columns: regions and TPMs"
+                )
+            if len(data) == 0:
+                raise ValueError(
+                    "CAGE TPM file is empty."
+                )
+            data.columns = ["regions", "CAGE"]
+
+            regions_split = data["regions"].str.split('[:-]', expand=True)
+            regions_split.columns = ["chrom", "start", "end"]
+            regions_split["start"] = regions_split["start"].astype(int)
+            regions_split["end"] = regions_split["end"].astype(int)
+            center = ((regions_split["start"] + regions_split["end"]) / 2).astype(int)
+            regions_split["start"] = center - 100
+            regions_split["end"] = center + 100
+
+            # Create region .bed file from CAGE input
+            force_rerun = True
+            regions_bed = f"{os.path.join(PACKAGE_DIR, 'db')}/regions.bed"
+            if force_rerun or not os.path.exists(regions_bed):
+                regions_split.to_csv(regions_bed, header=False, index=False, sep="\t")
+
+            # Determine average ReMap coverage (Currently only for genome = "hg19" or "hg38")
+            coverage_bw_path = ""
+            if "hg19" in self.genome:
+                coverage_bw_path = f"{os.path.join(PACKAGE_DIR, 'db')}/remap2022.hg19.w50.bw"
+            if "hg38" in self.genome:
+                coverage_bw_path = f"{os.path.join(PACKAGE_DIR, 'db')}/remap2022.hg38.w50.bw"
+
+            if len(coverage_bw_path) > 0:
+                self.region_type = "CAGE"  # Used later for selecting model type
+
+                logger.info("Determining average peak coverage")
+                remap_cov = coverage_table(peakfile=regions_bed, datafiles={coverage_bw_path}, window=200)
+                remap_cov = remap_cov.set_index(data["regions"])
+                remap_cov.rename(columns={remap_cov.columns[0]: "average"}, inplace=True)
+                self._avg = remap_cov
+                self._avg.columns = ["average"]
+                self._avg["average"] = self._avg["average"] / self._avg["average"].max()
```

Comment:
I could be wrong here, but can't we run this command once for all regions, then upload these files in the ANANSE db?

Then the regions only need to be filtered for each new run.